mastering/editable_packages.rst

```diff
@@ -0,0 +1,131 @@
+*editable packages* will do for you.
+
+
+Let's see this feature over an example where a developer is creating a ``CoolApp`` but at the same
+time he/she wants to work on ``cool/version@user/dev`` which is tightly coupled to the app.
+
+
+Setting it all up
+-----------------
+
+The package ``cool/version@user/dev`` is already working, the developer has the sources in a
+local folder, he/she is using whatever method to build and develop locally and can perform
+a :command:`conan create . libCool/version@user/dev` to create the package.
+
+Also, there is a *conanfile.txt* (or a more complex recipe) for the application ``CoolApp`` that
+has ``cool/version@user/dev`` among its requirements. When building this application, the
+resources of ``cool`` are used from the *data* directory inside the Conan local cache.
+
+
+Put a package in editable mode
+------------------------------
+
+To avoid creating the package ``cool/version@user/dev`` in the cache for every change, we can
+create **a link from the package in the cache to the local working directory**:
+
+.. code-block:: bash
+
+    $ conan link <path/to/local/dev/libcool> cool/version@user/dev
+
+and we need a file to tell Conan what is the layout of your local project in order to provide the
+proper paths to the build system. There are several ways to achieve this objective:
+
+ * A file next to the *conanfile.py* with the name *.conan_layout*:
+
+   .. code-block:: ini
+
+       [includedirs]
+       src/core/include
+       src/cmp_a/include
+
+       [libdirs]
+       build/{settings.build_type}/{options.shared}
+
+       [bindirs]
+       build/{settings.build_type}/{options.shared}
+
+ * A file inside the Conan cache in the path *layouts/default*. As this file can contain information
+   for several packages, each section must declare which packages it applies to, this is achieved
+   using a *namespace* with the name of the package or the wildcard ``*``:
+
+   .. code-block:: ini
+
+       [cool:includedirs]
+       src/core/include
+       src/cmp_a/include
+
+       [*:libdirs]
+       build/{settings.build_type}/{options.shared}
+
+       [*:bindirs]
+       build/{settings.build_type}/{options.shared}
+
+
+   This file can be very handy inside a company where all the packages have the same layout.
+
+As you can see, you can use inside these files some **placeholders** that will be substituted with
+the values of the ``settings`` and the ``options`` of the package.
+
+About the precedence if both files are present, Conan will use the one inside the repository if
+present, then the one in the Conan cache, and if none is available, Conan will fallback to the
+directories defined in the ``package_info`` method of the recipe itself but related to the path
+where the editable package has been linked to.
+
+
+Using a package in editable mode
+--------------------------------
+
+Once a package is in editable mode it is used **system wide** by Conan (by every Conan client that
```

Comment:
I think here you can also say that the editable binaries will be used regardless of the profile settings (unless you use hte suitable placeholders)

Reply:
Is it not understood from the next sentence: _"Every `conan install`
command that requires our editable ``cool/version@user/dev`` package will use the paths to
the local directory"_

Reply:
I understood it but I suspect we are a bit biased and it is better to be explicit

Reply:
I think that bias is thinking about different profile settings 😆 

But, yes, I will add a line there